### PR TITLE
Add spot and blob detection functions.

### LINKED ISSUE #114: Add spot-detection utils
* h-maxima
* DoG

see: https://github.com/fmi-faim/faim-hcs/tree/spot-utils

### PR DESCRIPTION
**Pull request by tibuch**
They combine a LoG detector with h-maxima.

Closes #114.

### CODE REVIEW

**Comment by tibuch** [comment]
Closes #114.

File: src/faim_ipa/detection/blobs.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    h :
+        h-maxima threshold.
+    n_scale_levels :
+        Number of upscaling rounds.
+    overlap :
+        A value between 0 and 1. If the fraction of area overlapping for 2
+        blobs is greater than `overlap` the smaller blob is eliminated.
+    background_img :
+        Estimated background image. This is subtracted before the
+        blob-detection.
```
Comment (imagejan) [inline src/faim_ipa/detection/blobs.py:46]:
```suggestion
        blob detection.
```

File: src/faim_ipa/detection/blobs.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
+from typing import Optional
+
+import numpy as np
+from scipy.ndimage import gaussian_laplace
+from skimage.feature import peak_local_max
+from skimage.morphology import h_maxima, ball
+from skimage.util import img_as_float32
+from skimage.feature.blob import _prune_blobs
```
Comment (imagejan) [inline src/faim_ipa/detection/blobs.py:8]: This is a private function, how about API stability? Is there some public alternative?

File: src/faim_ipa/detection/spots.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        Image containing spot signal.
+    axial_sigma :
+        Z extension of the spots.
+    lateral_sigma :
+        YX extension of the spots.
+    h :
+        h-maxima threshold.
+    background_img :
+        Estimated background image. This is subtracted before the
+        spot-detection.
```
Comment (imagejan) [inline src/faim_ipa/detection/spots.py:36]:
```suggestion
        spot detection.
```

**Review by imagejan — COMMENTED** [review]
Thanks @tibuch!
I just added a few nitpicky comments.

File: src/faim_ipa/detection/blobs.py (hunk 2 above)
Comment (tibuch) [reply src/faim_ipa/detection/blobs.py:8]: I agree, this is a bit unfortunate. But I couldn't find a public alternative. This function is used inside the blob detector in scikit-image. I am hoping that this is pretty stable and won't change.

Comment (tibuch) [reply src/faim_ipa/detection/blobs.py:8]: We could copy-paste the function out of skimage.

**Pull request merged**